apps/web/src/components/utils/Spotlight.tsx

```diff
@@ -1,67 +1,106 @@
-    addItem([
+  const navigateTo = useCallback(
+    (path) => {
+      navigate(path);
+    },
+    [navigate]
+  );
+
+  const openExternalLink = useCallback((url) => {
+    window?.open(url, '_blank')?.focus();
+  }, []);
+
+  const spotlightItems = useMemo(() => {
+    const buildEnvironmentTitle = (name) => `Toggle to ${getDestinationEnvironment(name)} environment`;
+
+    const buildEnvironmentOnTrigger = (name) => {
+      /**
+       * TODO: This setter doesn't work properly because internally the use-env-controller has
+       * isLoadingMyEnvironments=true and isLoadingCurrentEnvironment=true
+       */
+      setEnvironment(getDestinationEnvironment(name));
+    };
+
+    return [
       {
         id: 'navigate-templates',
         title: 'Go to Notification Template',
-        onTrigger: () => navigate('/templates'),
+        onTrigger: () => navigateTo('/templates'),
         icon: <Bolt />,
       },
       {
         id: 'navigate-integration',
         title: 'Go to Integrations',
-        onTrigger: () => navigate('/integrations'),
+        onTrigger: () => navigateTo('/integrations'),
         icon: <Box />,
       },
       {
         id: 'navigate-changes',
         title: 'Go to Changes',
-        onTrigger: () => navigate('/changes'),
+        onTrigger: () => navigateTo('/changes'),
         icon: <Repeat />,
       },
       {
         id: 'navigate-settings',
         title: 'Go to Settings',
-        onTrigger: () => navigate('/settings'),
+        onTrigger: () => navigateTo('/settings'),
         icon: <Settings />,
       },
       {
         id: 'navigate-activities',
         title: 'Go to Activities',
-        onTrigger: () => navigate('/activities'),
+        onTrigger: () => navigateTo('/activities'),
         icon: <Activity />,
       },
       {
         id: 'navigate-team-members',
         title: 'Go to Team Members',
-        onTrigger: () => navigate('/team'),
+        onTrigger: () => navigateTo('/team'),
         icon: <Team />,
       },
       {
         id: 'navigate-docs',
         title: 'Go to Documentation',
-        onTrigger: () => {
-          window?.open('https://docs.novu.co/', '_blank')?.focus();
-        },
+        onTrigger: () => openExternalLink('https://docs.novu.co/'),
       },
       {
         id: 'navigate-support',
         title: 'Go to Support',
-        onTrigger: () => {
-          window?.open('https://discord.com/invite/novu', '_blank')?.focus();
-        },
+        onTrigger: () => openExternalLink('https://discord.com/invite/novu'),
       },
-    ]);
-  }, []);
+      {
+        id: 'toggle-environment',
+        title: buildEnvironmentTitle(environmentName),
+        onTrigger: () => buildEnvironmentOnTrigger(environmentName),
+      },
+    ];
+  }, [environmentName, setEnvironment, navigateTo, openExternalLink]);
+
+  useEffect(() => {
+    if (items.length === 0) {
+      addItem(spotlightItems);
+    }
+  }, [addItem, spotlightItems, items]);
```

Comment:
Having to do that as the re-renders kept adding and adding items causing to reach the max update limit. Also causing to re-render the whole Web app as this provider was high in the hierarchy.